parser/match_operations.go

```diff
@@ -437,16 +437,41 @@ func comparisonMatch(
 		if err := matchIndexValid(matches, amountMatch[1]); err != nil {
 			return fmt.Errorf("%w: opposite amounts comparison error", err)
 		}
-		for _, op := range matches[amountMatch[0]].Operations {
-			for _, otherOp := range matches[amountMatch[1]].Operations {
-				if err := oppositeAmounts(
-					op,
-					otherOp,
-				); err != nil {
-					return fmt.Errorf("%w: amounts not opposites", err)
-				}
+
+		match1Ops := matches[amountMatch[0]].Operations
+		match2Ops := matches[amountMatch[1]].Operations
+
+		if len(match1Ops) != len(match2Ops) {
```

Comment:
nit: there is no expectation that `match1Ops` and `match2Ops` must be equal length

Reply:
Interesting. It does not make sense to me from a caller's perspective that there are different num of Pos & Neg operations with in a single Transaction. 

Or maybe this method is not restricted to the use case of a transaction? It just doing the comparison of the all amounts and make sure they are all opposite amounts, and it's up to the caller to determine what to do when they are unequal length matches in match result.

I think it makes sense if the above is true.